README.md

```diff
@@ -368,6 +383,19 @@ weakMap.set(record, true);
 
 # FAQ
 
+## What are the performance expectations of those Data Structures?
+
+This proposal in itself does not put any performance guarantees and does not require specific optimizations on the implementers. It is however built in a way that some performance optimizations can be done in most cases if implementers choose to do so.
```

Comment:
Map and Set impose specific requirements on implementations - namely, that retrieval must be sublinear as the size of the collection increases - why not do the same here?

Reply:
This is more like strings. it's not really possible to guarantee both constant time concatenation and indexing at once, but implementations try to make it kind of work anyway. We should explain this explicitly. This could also be a good place to paste the text I wrote in an issue about this topic.

Reply:
just pasted the text you wrote, I just had to retrieve it